src/vm/eetwain.cpp

```diff
@@ -3805,6 +3806,28 @@ bool UnwindEbpDoubleAlignFrame(
     {
         TADDR baseSP;
 
+#ifdef WIN64EXCEPTIONS
+        // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+        // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+        // If funclet frame layout is changed from CodeGen::genFuncletProlog() and genFuncletEpilog(),
+        // we need to change here accordingly.
+        if (pCodeInfo->IsFunclet())
+        {
+            baseSP = curESP + 8; // padding for 16byte stack alignment allocated in genFuncletProlog()
```

Comment:
@parjong a function can have multiple regions of interruptibility / non-interruptibility. You can see that from a jitdump of a function.
You can see something like this by the end of the jitdump (COMPlus_JitDump=methodname) output:
Defining interruptible range: [0x1b, 0xdc).
Defining interruptible range: [0xf2, 0x136).
Defining interruptible range: [0x146, 0x17e).
This comes from an old jitdump that I have captured about a year ago when debugging something. See the code of a funclet below that's related to the last of the interruptible ranges. You can see that the whole funclet is interruptible except for its prologue and epilogue.
You can generate a jitdump for a function that has try / catch on x86 to see which regions are interruptible.
```
G_M22228_IG11:        ; func=02, offs=000136H, size=0010H, gcVars=0000000000007F49 {V00 V02 V03 V05 V06 V07 V08 V11 V12 V13}, gcrefRegs=00000040 {rsi}, byrefRegs=00000000 {}, gcvars, byref, funclet prolog, nogc

IN005a: 000136 push     rbp
IN005b: 000137 sub      rsp, 16
IN005c: 00013B mov      rbp, qword ptr [rdi]
IN005d: 00013E mov      qword ptr [rsp], rbp
IN005e: 000142 lea      rbp, [rbp+70H]

G_M22228_IG12:        ; offs=000146H, size=0038H, gcVars=0000000000007F49 {V00 V02 V03 V05 V06 V07 V08 V11 V12 V13}, gcrefRegs=00000040 {rsi}, byrefRegs=00000000 {}, gcvars, byref, isz

IN003c: 000146 mov      gword ptr [V09 rbp-48H], rsi
IN003d: 00014A mov      rsi, gword ptr [V09 rbp-48H]
IN003e: 00014E mov      gword ptr [V03 rbp-20H], rsi
IN003f: 000152 nop      
IN0040: 000153 mov      rsi, 0x7F00100010F0
IN0041: 00015D mov      rsi, gword ptr [rsi]
IN0042: 000160 mov      rdi, gword ptr [V00 rbp-10H]
IN0043: 000164 call     System.String:Concat(ref,ref):ref
IN0044: 000169 mov      gword ptr [V10 rbp-50H], rax
IN0045: 00016D mov      rsi, gword ptr [V10 rbp-50H]
IN0046: 000171 mov      gword ptr [V00 rbp-10H], rsi
IN0047: 000175 nop      
IN0048: 000176 nop      
IN0049: 000177 lea      rax, G_M22228_IG06

G_M22228_IG13:        ; offs=00017EH, size=0006H, funclet epilog, nogc, emitadd

IN005f: 00017E add      rsp, 16
IN0060: 000182 pop      rbp
IN0061: 000183 ret 
```